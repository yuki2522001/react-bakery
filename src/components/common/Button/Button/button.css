.btn {
  color: var(--text-master);
  border-color: var(--text-master);
  border-radius: 3px;
  background-color: transparent;
  cursor: pointer;
  font-family: "Poppins";
  font-weight: bold;
}

.btn__primary {
  display: flex;
  align-self: center;
  line-height: 30px;
  height: 35px;
}

.btn__secondary {
  padding: 5px;
  margin: 0 auto;
  align-self: center;
  display: flex;
}

.btn__secondary:hover,
.btn__info:hover {
  color: var(--btn-primary);
  background-color: var(--bg-view);
}

.btn__info {
  margin: 20px;
}

.btn__update {
  border: none;
  color: var(--bg-submit);
  height: 35px;
  font-size: larger;
}

.btn__delete {
  position: absolute;
  z-index: 5;
  top: 3%;
  right: 5%;
  color: var(--bg-delete);
  border: none;
  font-size: 15px;
  display: none;
}

.btn__delete--storybook {
  color: var(--bg-delete);
  border: none;
  font-size: 15px;
}

.btn__delete:hover {
  display: block;
}

.btn__warning,
.btn__success {
  border: 1px solid var(--border-btn);
  margin: 3px;
}

.btn__success:hover {
  color: var(--btn-primary);
  border-color: var(--bg-submit);
  background-color: var(--bg-submit);
}

.btn__warning:hover {
  color: var(--btn-primary);
  border-color: var(--bg-cancel);
  background-color: var(--bg-cancel);
}