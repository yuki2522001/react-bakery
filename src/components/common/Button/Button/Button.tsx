import React, { memo } from "react";
import { ButtonProps } from "@common-types/button";
import "./button.css";

const Button: React.FC<ButtonProps> = memo(
  ({ text, type, onClick, disabled }) => {
    let className = "btn";
    switch (type) {
      case "primary":
        className += " btn__primary";
        break;
      case "secondary":
        className += " btn__secondary";
        break;
      case "info":
        className += " btn__info";
        break;
      case "update":
        className += " btn__update";
        break;
      case "delete":
        className += " btn__delete";
        break;
      case "warning":
        className += " btn__warning";
        break;
      case "success":
        className += " btn__success";
        break;
      default:
        break;
    }
    return (
      <button className={className} disabled={disabled} onClick={onClick}>
        {text}
      </button>
    );
  }
);

export default Button;
