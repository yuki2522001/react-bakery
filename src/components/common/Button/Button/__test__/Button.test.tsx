import { fireEvent, render, screen } from "@testing-library/react";
import Button from "../Button";

describe("Button component", () => {
  const onClickButton = jest.fn();
  test("Should render prop when clicking button", async () => {
    render (
      <Button onClick={onClickButton} text="Submit" />
    );
    const clickBtn = screen.getByRole("button", {name: /Submit/i});
    fireEvent.click(clickBtn);
    expect(onClickButton).toHaveBeenCalled();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<Button text="" onClick={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });
});