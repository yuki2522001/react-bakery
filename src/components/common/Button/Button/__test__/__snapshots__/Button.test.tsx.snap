// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn"
  />
</DocumentFragment>
`;
